go-controller/pkg/ovn/controller/admin_network_policy/repair.go

```diff
@@ -0,0 +1,281 @@
+package adminnetworkpolicy
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/ovsdb"
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	libovsdbutil "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/util"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+	anpapi "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+)
+
+// repairAdminNetworkPolicies is called at startup and as the name suggests
+// aims to repair the NBDB logical objects -> port groups, acls and address-sets
+// that are created for the admin network policies in the cluster
+// Logic:
+// We fetch all the ANPs present in the cluster from the Lister
+// We fetch PGs, ACLs, AddressSets that are owned by ANP objectIDs based on
+// externalIDs match from the NBDB. Using predicate search we check if
+// the relevant ANP still exists for these objects and if not we delete them
+// TODO (tssurya): Handle cleaning up stale IPs in the address-sets by fetching all the relevant
+// peer pods that are being served by ANPs - pods could stop matching when ovnkube-controller is down
+// TODO2 (tssurya): Handle cleaning up stale ports from PGs by fetching all the relevant
+// subject pods that are being served by ANPs - pods could stop matching when ovnkube-controller is down
+func (c *Controller) repairAdminNetworkPolicies() error {
+	start := time.Now()
+	defer func() {
+		klog.Infof("Repairing admin network policies took %v", time.Since(start))
+	}()
+	c.Lock()
+	defer c.Unlock()
+	anps, err := c.anpLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("unable to list ANPs from the lister, err: %v", err)
+	}
+	existingANPs := map[string]*anpapi.AdminNetworkPolicy{}
+	for _, anp := range anps {
+		existingANPs[anp.Name] = anp
+		// let's populate the anpPriorityMap cache and add the correct ANP at the right priority
+		if anp.Spec.Priority > ovnkSupportedPriorityUpperBound {
+			// we don't want to add this ANP to the cache since we don't support this priority range
+			continue
+		}
+		ovnPriority := ANPFlowStartPriority - anp.Spec.Priority*ANPMaxRulesPerObject
+		status := meta.FindStatusCondition(anp.Status.Conditions, policyReadyStatusType+c.zone)
+		if status != nil && strings.Contains(status.Message, "exists with the same priority") {
```

Comment:
Perhaps use the Error or a constant for `exists with the same priority`

BTW, I guess it would be possible for two policies with the same priority to be added at the same time, and that two different zones consider two different policies.

Also if there is two duplicate policies, and the one considered is deleted, the other one will stay in error.

We ultimately need an admission controller for this?